realm/realm-library/src/objectServer/java/io/realm/internal/async/RealmResultTaskImpl.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm.internal.async;
+
+import java.util.concurrent.Future;
+import java.util.concurrent.ThreadPoolExecutor;
+
+import io.realm.internal.RealmNotifier;
+import io.realm.internal.Util;
+import io.realm.internal.android.AndroidCapabilities;
+import io.realm.internal.android.AndroidRealmNotifier;
+import io.realm.log.RealmLog;
+import io.realm.mongodb.App;
+import io.realm.mongodb.AppException;
+import io.realm.mongodb.ErrorCode;
+import io.realm.mongodb.RealmResultTask;
+
+/**
+ * Specific implementation of {@link RealmResultTask}.
```

Comment:
This description doesn't really say anything you cannot tell from just reading the class name. Perhaps something like `Implementation of RealmResultTask used internally by MongoDB Realm API's. Implementation is separate from the interface so we can hide the constructor from end users.`